tools/bundle-open-api-tpl-files.ts

```diff
@@ -0,0 +1,262 @@
+};
+
+if (isRunningDirectlyViaCLI) {
+  main(process.argv, process.env);
+}
+
+async function createRequest(
+  argv: string[],
+  env: NodeJS.ProcessEnv,
+): Promise<IBundleOpenApiTplFilesV1Request> {
+  if (!argv) {
+    throw new RuntimeError(`Process argv cannot be falsy.`);
+  }
+  if (!env) {
+    throw new RuntimeError(`Process env cannot be falsy.`);
+  }
+
+  const __filename = fileURLToPath(import.meta.url);
+  const __dirname = path.dirname(__filename);
+  const SCRIPT_DIR = __dirname;
+  const DEFAULT_WORKING_DIRECTORY = path.join(SCRIPT_DIR, "../");
+
+  const DEFAULT_GLOB = "**/src/main/json/openapi.tpl.json";
+
+  const DEFAULT_IGNORE_GLOB = "**/node_modules";
+
+  const optCheckGitDiffDisabled =
+    "This feature is not yet implemented. Check back later!";
+
+  const optDescriptionSpecTemplateGlobs =
+    "The GLOB pattern(s) to use when locating the openapi.tpl.json " +
+    "files from which to render the bundled final output with " +
+    "the filename of openapi.json. Defaults to: [" +
+    DEFAULT_GLOB +
+    "]";
+
+  const optDescriptionSpecTemplateIgnoreGlobs =
+    "The **IGNORE** GLOB pattern(s) to use when locating the " +
+    "openapi.tpl.json files from which to render the bundled final output " +
+    "with the filename of openapi.json. Defaults to: [" +
+    DEFAULT_IGNORE_GLOB +
+    "]";
+
+  const parsedCfg = await yargs(hideBin(argv))
+    .env("CACTI_")
+    .usage("yarn tools:bundle-open-api-tpl-files")
+    .option("verbose", {
+      alias: "v",
+      type: "boolean",
+      description: "Whether logs should be sent to stdout/stderr or not.",
+      default: false,
+    })
+    .option("working-directory", {
+      alias: "d",
+      type: "string",
+      description:
+        "The current working directory to search for openapi.tpl.json files for.",
+      default: DEFAULT_WORKING_DIRECTORY,
+    })
+    .option("check-git-diff-disbled", {
+      alias: "c",
+      type: "boolean",
+      description: optCheckGitDiffDisabled,
+      default: true,
+    })
+    .option("spec-template-globs", {
+      alias: "s",
+      type: "array",
+      string: true,
+      description: optDescriptionSpecTemplateGlobs,
+      default: [DEFAULT_GLOB],
+    })
+    .option("spec-template-ignore-globs", {
+      alias: "i",
+      type: "array",
+      string: true,
+      description: optDescriptionSpecTemplateIgnoreGlobs,
+      default: [DEFAULT_IGNORE_GLOB],
+    })
+    .version("v0.0.1")
+    .parse();
+
+  const req: IBundleOpenApiTplFilesV1Request = {
+    argv,
+    checkGitDiffDisabled: parsedCfg.checkGitDiffDisbled,
+    specTemplateGlobs: parsedCfg.specTemplateGlobs,
+    specTemplateIgnoreGlobs: parsedCfg.specTemplateIgnoreGlobs,
+    verbose: parsedCfg.verbose,
+    workingDirectory: parsedCfg.workingDirectory,
+  };
+
+  return req;
+}
+
+function log(
+  req: IBundleOpenApiTplFilesV1Request,
+  message: string,
+  ...args: unknown[]
+): void {
+  if (req.verbose) {
+    console.log(message, ...args);
```

Comment:
Could you use some 3'rd party library for logging and set `DEBUG` when verbose flag was provided? This would allow more granular messaging in the future (in case script causes any troubles and need to be tampered with in the future).

Reply:
@outSH Thank you for the great suggestion! I've made the changes just now, it uses the library called "debug"